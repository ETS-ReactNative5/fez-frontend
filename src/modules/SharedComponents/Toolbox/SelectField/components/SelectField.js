import React from 'react';
import Select from '@material-ui/core/Select';
import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import FormHelperText from '@material-ui/core/FormHelperText';

import PropTypes from 'prop-types';
import propFilter from '../../helpers/_filterProps';

const SelectFieldWrapper = props => {
    const filteredProps = propFilter({...props, forceError: true}, Select.propTypes);
    filteredProps.onChange = (event) => props.input.onChange(event.target.value);
    filteredProps.onBlur = () => props.input.onBlur(props.input.value);
    const error = !!filteredProps.errorText || !!filteredProps.error;
    const helperText = filteredProps.errorText || filteredProps.error || null;
    delete filteredProps.errorText;
    return (
        <React.Fragment>
            <FormControl error={error} style={{width: '100%'}} required={filteredProps.required}>
                <InputLabel>{filteredProps.label}</InputLabel>
                <Select {...filteredProps} autoWidth />
                {
                    helperText &&
                    <FormHelperText>{helperText}</FormHelperText>
                }
            </FormControl>
        </React.Fragment>
    );
};

SelectFieldWrapper.propTypes = {
    ...Select.propTypes,
    help: PropTypes.shape({
        title: PropTypes.string,
        text: PropTypes.any,
        buttonLabel: PropTypes.string
    })
};

export default SelectFieldWrapper;
