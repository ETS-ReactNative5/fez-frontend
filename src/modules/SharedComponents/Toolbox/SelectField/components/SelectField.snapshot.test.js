import SelectField from './SelectField';

import filterProps from '../../helpers/_filterProps';

function setup(testProps, isShallow = true) {
    const props = {
        name: 'selectfield',
        type: 'text',
        fullWidth: true,
        floatingLabelText: 'This is a test selectfield component',
        ...testProps
    };
    const consolidatedProps = filterProps(props, SelectField.propTypes);
    return getElement(SelectField, consolidatedProps, isShallow);
}

describe('SelectfieldWrapper snapshots tests', () => {
    it('renders SelectField component', () => {
        const props =
            {
                name: 'selectfield',
                type: 'text',
                fullWidth: true,
                floatingLabelText: 'This is a test selectfield component'
            };

        const wrapper = setup(props);
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});
