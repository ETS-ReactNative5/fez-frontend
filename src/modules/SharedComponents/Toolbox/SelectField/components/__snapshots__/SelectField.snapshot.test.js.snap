// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectfieldWrapper snapshots tests renders SelectField component 1`] = `
<React.Fragment>
  <WithStyles(FormControl)
    error={false}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <WithStyles(InputLabel) />
    <WithStyles(Select)
      autoWidth={true}
      error={false}
      floatingLabelText="This is a test selectfield component"
      fullWidth={true}
      name="selectfield"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
  </WithStyles(FormControl)>
</React.Fragment>
`;
